src/drivers/camera_trigger/interfaces/src/gpio.cpp

```diff
@@ -22,8 +22,11 @@ CameraInterfaceGPIO::~CameraInterfaceGPIO()
 void CameraInterfaceGPIO::setup()
 {
 	for (unsigned i = 0; i < arraySize(_pins); i++) {
-		px4_arch_configgpio(_gpios[_pins[i]]);
-		px4_arch_gpiowrite(_gpios[_pins[i]], !_polarity);
+		// Pin range is ranges from 1 to 6
+		if (_pins[i] > 0 && _pins[i] < (int)arraySize(_gpios)) {
```

Comment:
From what I see this should be `_pins[i] >= 0 && ...`
Note the check in `CameraInterfaceGPIO::trigger`: it uses `if (_pins[i] >= 0) {`

Reply:
@bkueng Could you provide a final answer here?

Reply:
So do we need to change this or not?

Reply:
Yes, needs a change. `_pins[i]` is set to `param value -1` in https://github.com/PX4/Firmware/blob/6f7b17d693a16ebb7a89210d2a7166f5f7c44b08/src/drivers/camera_trigger/interfaces/src/camera_interface.cpp#L42
and param value == 1 is a valid setting. Fix pushed.